worker/worker.go

```diff
@@ -519,6 +519,47 @@ func (w *worker) rhpScanHandler(jc jape.Context) {
 	})
 }
 
+func (w *worker) fetchActiveContracts(ctx context.Context, metadatas []api.ContractMetadata, timeout time.Duration, bh uint64) (contracts []api.Contract, errs HostErrorSet) {
+	// fill requests channel
+	reqs := make(chan api.ContractMetadata, len(metadatas))
+	for _, metadata := range metadatas {
+		reqs <- metadata
+	}
+	close(reqs)
+
+	// create worker function
+	var mu sync.Mutex
+	worker := func() {
+		for metadata := range reqs {
+			rev, err := w.FetchRevision(ctx, timeout, metadata, bh, lockingPriorityActiveContractRevision, lockingDurationActiveContractRevision)
+			mu.Lock()
+			if err != nil {
+				errs = append(errs, &HostError{HostKey: metadata.HostKey, Err: err})
+			} else {
+				contracts = append(contracts, api.Contract{
+					ContractMetadata: metadata,
+					Revision:         rev,
+				})
+			}
+			mu.Unlock()
+		}
+	}
+
+	// launch all workers
+	var wg sync.WaitGroup
+	for t := 0; t < 10 && t < len(metadatas); t++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			worker()
+		}()
+	}
```

Comment:
Why not move the filling below this line and avoid making the channel buffered?

Reply:
You really hate buffered channels 😁